src/engine/strat_engine/backstore/metadata/mda.rs

```diff
@@ -0,0 +1,581 @@
+                format!("Unknown metadata version: {}", metadata_version),
+            ));
+        }
+
+        match LittleEndian::read_u64(&buf[16..24]) {
+            0 => Ok(None),
+            secs => {
+                let used = Bytes(LittleEndian::read_u64(&buf[8..16]));
+
+                // Signed cast is safe, highest order bit of each value
+                // read is guaranteed to be 0.
+                assert!(secs <= std::i64::MAX as u64);
+
+                let nsecs = LittleEndian::read_u32(&buf[24..28]);
+                Ok(Some(MDAHeader {
+                    used,
+                    last_updated: Utc.timestamp(secs as i64, nsecs),
+                    data_crc: LittleEndian::read_u32(&buf[4..8]),
+                }))
+            }
+        }
+    }
+
+    fn to_buf(&self) -> [u8; mda_size::_MDA_REGION_HDR_SIZE] {
+        // Unsigned casts are always safe, as sec and nsec values are never negative
+        assert!(self.last_updated.timestamp() >= 0);
+
+        let mut buf = [0u8; mda_size::_MDA_REGION_HDR_SIZE];
+
+        LittleEndian::write_u32(&mut buf[4..8], self.data_crc);
+        LittleEndian::write_u64(&mut buf[8..16], *self.used as u64);
+        LittleEndian::write_u64(&mut buf[16..24], self.last_updated.timestamp() as u64);
+        LittleEndian::write_u32(&mut buf[24..28], self.last_updated.timestamp_subsec_nanos());
+        buf[28] = STRAT_REGION_HDR_VERSION;
+        buf[29] = STRAT_METADATA_VERSION;
+
+        let buf_crc = crc32::checksum_castagnoli(&buf[4..mda_size::_MDA_REGION_HDR_SIZE]);
+        LittleEndian::write_u32(&mut buf[..4], buf_crc);
+
+        buf
+    }
+
+    /// Given a pre-seek()ed File, load the MDA region and return the contents.
+    /// Return an error if the data can not be read, since the existence
+    /// of the MDAHeader implies that the data must be available.
+    // MDAHeader cannot seek because it doesn't know which region it's in
+    fn load_region<F>(&self, f: &mut F) -> StratisResult<Vec<u8>>
+    where
+        F: Read,
+    {
+        // This cast could fail if running on a 32-bit machine and
+        // size of metadata is greater than 2^32 - 1 bytes, which is
+        // unlikely.
+        //
+        // This comparison seems absurd when compiled in an environment
+        // where usize is u64, which is usual. It is not absurd when
+        // compiled in an environment where usize is u32.
+        #![allow(clippy::absurd_extreme_comparisons)]
+        assert!(*self.used <= std::usize::MAX as u64);
+        let mut data_buf = vec![0u8; *self.used as usize];
+
+        f.read_exact(&mut data_buf)?;
+
+        if self.data_crc != crc32::checksum_castagnoli(&data_buf) {
+            return Err(StratisError::Engine(
+                ErrorEnum::Invalid,
+                "MDA region data CRC".into(),
+            ));
+        }
+
+        Ok(data_buf)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::io::Cursor;
+
+    use super::super::bda::BDA_STATIC_HDR_SIZE;
```

Comment:
Whoops! I missed a spot. Could you please change this to a crate import as well, as ```use crate::engine::strat_engine::backstore::metadata::bda::BDA_STATIC_HDR_SIZE;``` and move it down between the external crate imports and ```use super::*;```. We place imports in order of specificity: standard library, external crates, stratisd's own, and then, and only in tests, ```use super::*````.